nginx/nginx.conf

```diff
@@ -0,0 +1,83 @@
+        server haproxy:5984;
+    }
+
+    server {                                         
+        listen         80;
+        server_name    _;
+    #    error_log /etc/nginx/logs/error.log debug;
+        location / {                                                           
+            return 301 https://$host$request_uri;
+        }                                                      
+    }                                                                          
+                                                                               
+    server {                                                                   
+        listen       443 ssl;
+        server_name  _;
+    #    error_log /etc/nginx/logs/error-ssl.log debug;
+    #    access_log /etc/nginx/logs/access-ssl.log;
+                                                                               
+        gzip                on;
+        gzip_disable        "msie6";
+        gzip_vary           on;
+        gzip_proxied        any;
+        gzip_comp_level     6;
+        gzip_buffers        16 8k;
+        gzip_http_version   1.1;
+
+        gzip_types                  text/plain text/css text/csv text/xml text/javascript
+                                    application/json application/x-javascript application/xml
+                                    application/vnd.ms-fontobject application/octet-stream
+                                    application/x-font-woff multipart/related image/svg+xml;
+
+        ssl_certificate             /etc/nginx/private/cert.pem;
+        ssl_certificate_key         /etc/nginx/private/key.pem;
+        ssl_prefer_server_ciphers   on;
+        ssl_session_timeout         30m;
+        ssl_session_cache           shared:SSL:5m;
+        ssl_protocols               TLSv1.2 TLSv1.1 TLSv1;
+        ssl_ciphers                 ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA:DHE-RSA-CAMELLIA256-SHA:DHE-RSA-AES128-SHA:DES-CBC3-SHA:!EXP:!ADH:!aNULL;
+
+        # Prevent "clickjacking" attacks:
+        #   This disallows external sites from embedding any of our pages in
+        #   an <iframe>. Since medic-reporter currently uses an <iframe>, we've
+        #   set the value to SAMEORIGIN instead of DENY. If we ever stop using
+        #   <iframes>s, we should probably switch this back to DENY.
+
+        add_header X-Frame-Options  SAMEORIGIN;
+
+        location / {
+            proxy_pass              http://medic-api:5988;
+            proxy_set_header        Host            $host;
+            proxy_set_header        X-Real-IP       $remote_addr;
+            proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header        X-Forwarded-Proto https;
+            proxy_set_header        X-Forwarded-Host $server_name;
+        }
+
+        error_page   500 502 503 504  /50x.html;
+
+        location = /50x.html {
+            root html;
+        }
+
+    }
+}
+
+# vim: set nowrap tw=0:
```

Comment:
Please remove this. I was going to ask if vim respects this comment, but even if it does, it still seems unusual.